tools/flang1/flang1exe/semfunc.c

```diff
@@ -11474,6 +11474,36 @@ ref_pd_subr(SST *stktop, ITEM *list)
     REFP(pdsym, 1);
     break;
 
+  case PD_getarg:
+    if (count != 2) {
+      E74_CNT(pdsym, count, 2, 2);
+      goto call_e74_cnt;
+    }
+    if (get_kwd_args(list, 2, KWDARGSTR(pdsym)))
+      goto exit_;
+    sp = ARG_STK(0); /* pos */
+    (void)mkexpr(sp);
+    XFR_ARGAST(0);
+    dtype2 = SST_DTYPEG(sp);
+    if (!DT_ISINT(dtype2)) {
+      E74_ARG(pdsym, 0, NULL);
+      goto call_e74_arg;
+    }
+    if ((sp = ARG_STK(1))) { /* value */
+      if (!is_varref(sp)) {
+        E74_ARG(pdsym, 1, NULL);
+        goto call_e74_arg;
+      }
+      (void)mkarg(sp, &dum);
+      XFR_ARGAST(1);
+      dtype2 = SST_DTYPEG(sp);
+      if (DTY(dtype2) != TY_CHAR) {
+        E74_ARG(pdsym, 1, NULL);
+        goto call_e74_arg;
+      }
+    }
```

Comment:
The processing of the first two arguments is identical to that in `case PD_get_command_argument`. Perhaps some code reuse is possible.

Reply:
Do you mean to refactor the similar blocks out as a function?

Reply:
Also, I'm not quite sure what is the exact requirement for the 1st arg of `GETARG`. In gfortran, it is illegal for the 1st argument to be of an integer kind wider than the default kind, while in ifort, it is ok. Do you have any suggestions?

Reply:
`GETARG` was not a standard Fortran intrinsic; vendors would define it differently. For its first argument, it was mostly declared to be the default `integer` (`integer` in [HP Fortran](http://binf.gmu.edu/jafri/HP_Fortran_Programmers_Reference.pdf), `integer*4` in [Sun Studio](https://docs.oracle.com/cd/E19205-01/819-5259/aetdh/index.html), `integer*2` in [Microsoft Fortran](http://computer-programming-forum.com/49-fortran/c2241492a04c9fe4.htm), etc.). I think gfortran accepts all `integer` types narrower than the default `integer` simply to support more legacy programs. The Intel behaviour is probably an extension, and I am not sure that it is all that valuable.

Reply:
About code reuse, yes, I thought you could factor out the blocks into a function. You could use labels and `goto` which is fairly common in the code base, but I don't recommend that practice any more. Leaving it as is fine as well.

Reply:
Never mind, I think factoring the code is probably not worthwhile.